doc/specs/#1595 - Suggestions UI/Suggestions-UI.md

```diff
@@ -0,0 +1,734 @@
+to be extended in a variety of ways. Those customizations will all be opt-in by
+the user, so I'm not expecting any breaking compatibility changes here.
+
+</td></tr>
+
+<tr><td><strong>Accessibility</strong></td><td>
+
+The Suggestions UI was designed with the goal of making commandline shell
+suggestions _more_ accessible. As Carlos previously wrote:
+
+> Screen readers struggle with this because the entire menu is redrawn every time, making it harder to understand what exactly is "selected" (as the concept of selection in this instance is a shell-side concept represented by visual manipulation).
+> 
+> ...
+> 
+> _\[Shell driven suggestions\]_ can then be leveraged by Windows Terminal to create UI elements. Doing so leverages WinUI's accessible design. 
+
+This will allow the Terminal to provide more context-relevant information to
+screen readers. 
+
+</td></tr>
+
+<tr><td><strong>Sustainability</strong></td><td>
+
+No sustainability changes expected.
+
+</td></tr>
+
+<tr><td><strong>Localization</strong></td><td>
+
+The localization needs of the Suggestions UI will be effectively the same as the
+needs of the Command Palette.
+
+The Terminal will have no way to localize suggestions that are provided via
+[shell-driven autocompletion]. These are just verbatim strings that the shell
+told us to use. We don't consider this to be something to worry about, however.
+This is no different than the fact that Terminal cannot localize the `Get-Help`
+(or any other) output of PowerShell.
+
+</td></tr>
+
+</table>
+
+## Implementation Plan
+
+This is more of an informative outline, rather than a normative one. Many of the
+things from Crawl, Walk, and Run are all already in PRs as of the time of this
+spec's review.
+
+### 🐣 Crawl
+
+* [ ] Fork the Command palette to a new UI element, the `SuggestionsControl`
+* [ ] Enable previewing `sendInput` actions in the Command Palette and `SuggestionsControl`
+* [ ] Enable the `SuggestionsControl` to open top-down (aligned to the bottom of the cursor row) or bottom-up (aligned to the top of the cursor row). 
+* [ ] Disable sorting on the `SuggestionsControl` - elements should presumably be pre-sorted by the source. 
+* [ ] Expose the recent commands as a accessor on `TermControl`
+* [ ] Add a `suggestions` action which accepts a single option `recentCommands`. These should be fed in MRU order to the `SuggestionsControl`.
+* [ ] Expose the recent directories as an accessor on `TermControl`, and add a `recentDirectories` source.
+
+### 🚶 Walk
+
+* [ ] Add a `tasks` source to `suggestions` which opens the Suggestions UI with
+  a tree of all `sendInput` commands
+* [ ] Enable the `SuggestionsControl` to open with or without a search box
+* [ ] Plumb support for shell-driven completions through the core up to the app
+* [ ] Figure out how exactly we want to persist recent commands / directories. Globally? Per-profile? Layered? Per executable (does that even make sense)?

```

Comment:
Actually yeah, we may want to promote this to be part of the spec proper. There's data we need to store; how do we store it?

Reply:
I see it in future considerations. If it's part of "Walk," it is a Now Consideration!